### PR TITLE
[V2] feat: enable controller pod customization

### PR DESCRIPTION
**Pull request by edreed**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Enables limited customization of the controller pod.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: charts/latest-v2/azuredisk-csi-driver/values.yaml [hunk 1]
```diff
@@ -135,6 +135,19 @@ controller:
       requests:
         cpu: 10m
         memory: 20Mi
+  environments:
```
Comment (sunpa93) [inline charts/latest-v2/azuredisk-csi-driver/values.yaml:138]: Are the values under environments used anywhere?

Comment (edreed) [reply charts/latest-v2/azuredisk-csi-driver/values.yaml]: This was from an early iteration. Removed.

**Pull request merged**